docs/rfcs/1011-non-blocking-invite-transport.md

```diff
@@ -0,0 +1,1226 @@
+        ]
+    }
+]
+```
+
+
+### Removed routes
+
+All the `invite_<N>_claimer_<step name>` and `invite_<N>_greeter_<step name>` are removed and replaced by `invite_claimer_step` and `invite_greeter_step`.
+
+This includes:
+- `invite_1_greeter_wait_peer`
+- `invite_2a_greeter_get_hashed_nonce`
+- `invite_2b_greeter_send_nonce`
+- `invite_3a_greeter_wait_peer_trust`
+- `invite_3b_greeter_signify_trust`
+- `invite_4_greeter_communicate`
+- `invite_1_claimer_wait_peer`
+- `invite_2a_claimer_send_hashed_nonce`
+- `invite_2b_claimer_send_nonce`
+- `invite_3a_claimer_signify_trust`
+- `invite_3b_claimer_wait_peer_trust`
+- `invite_4_claimer_communicate`
+
+
+### Protocol update
+
+Here is an overview of the different kinds of data that can be contained in a claimer or greeter step:
+- public key: `PublicKey`
+- nonce: `Bytes`
+- hashed nonce: `HashDigest`
+- acknowledgement: `None`
+- payload: `Bytes`
+
+The news steps are defined as such:
+
+| Index | Greeter step       | Greeter data   | Claimer step        | Claimer data   |
+|-------|--------------------|----------------|---------------------|----------------|
+| 0     | `wait_peer`        | public key     | `wait_peer`         | public key     |
+| 1     | `get_hashed_nonce` |                | `send_hashed_nonce` | hashed nonce   |
+| 2     | `send_nonce`       | nonce          | `get_nonce`         |                |
+| 3     | `get_nonce`        |                | `send_nonce`        | nonce          |
+| 4     | `wait_peer_trust`  |                | `signify_trust`     | acknowledgment |
+| 5     | `signify_trust`    | acknowledgment | `wait_peer_trust`   |                |
+| 6     | `get_payload`      |                | `send_payload`      | payload        |
+| 7     | `send_payload`     | payload        | `get_payload`       |                |
+| 8     | `wait_peer_ack`    |                | `acknowledge`       | acknowledgment |
+
+Those steps work very similarly to Parsec v2. There are a couple of differences though:
+- Steps are not longer counted as `1`, `2a`, `2b`, `3a`, `3b`, and `4` like in version 2
+- Three different exchanges were performed in steps `2a` and `2b`. Now those exchanges are expanded in steps `1`, `2` and `3`.
+- Step `4` in version 2 could be used for several exchanges, until the greeter marked an exchange with `last=True`. Instead, those exchanges are expanded in steps `6` and `7`
+- In version 2, the last step was used to complete the invite. In version 3 this is now done with a dedicated `invite_complete` command, so we add a last step (index `8`) for the claimer to acknowledge that it's been able to successfully process the greeter payload.
+
+The `GreeterStep` nested data type is then define as such:
+```json5
+            {
+                "name": "GreeterStep",
+                "discriminant_field": "step",
+                "variants": [
+                    {
+                        "name": "WaitPeer",
+                        "discriminant_value": "0",
```

Comment:
Note that `discriminant_value` are used as literal (as string not integer).
Maybe we could use the `name` as discriminant instead of `int` ?

Reply:
Yes I'm not fully decided on those definitions. However I think it's useful to have the number of step as constant somewhere.

Reply:
We would come back eventually to the hybrid method with both the name + index :sweat_smile:

Reply:
I also think it would be better to have name as `Step0WaitPeer`, otherwise it's too cumbersome to work with those types, especially given there is 8 of them !